Source/VolunteerReporting/TextMessaging/TextMessageProcessor.cs

```diff
@@ -62,84 +62,100 @@ public void Process(TextMessage message)
             {
                 var sex = (Sex)parsingResult.Numbers[1];
                 var age = parsingResult.Numbers[2];
-                ReportSingle(message, dataCollectorId, caseReporting, healthRiskId, sex, age, unknown);
+                var malesUnder5 = age <= 5 && sex == Sex.Male ? 1 : 0;
+                var malesOver5 = age > 5 && sex == Sex.Male ? 1 : 0;
+                var femalesUnder5 = age <= 5 && sex == Sex.Female ? 1 : 0;
+                var femalesOver5 = age > 5 && sex == Sex.Female ? 1 : 0;
+                Report(
+                    message,
+                    dataCollectorId,
+                    unknown,
+                    caseReporting,
+                    healthRiskId,
+                    malesUnder5,
+                    malesOver5,
+                    femalesUnder5,
+                    femalesOver5,
+                    message.Sent);
             }
             else
             {
                 var malesUnder5 = parsingResult.Numbers[1];
                 var malesOver5 = parsingResult.Numbers[2];
                 var femalesUnder5 = parsingResult.Numbers[3];
                 var femalesOver5 = parsingResult.Numbers[4];
-
-                ReportMultiple(message, dataCollectorId, unknown, caseReporting, healthRiskId, malesUnder5, malesOver5, femalesUnder5, femalesOver5);
+                Report(
+                    message,
+                    dataCollectorId,
+                    unknown,
+                    caseReporting,
+                    healthRiskId,
+                    malesUnder5,
+                    malesOver5,
+                    femalesUnder5,
+                    femalesOver5,
+                    message.Sent);
             }
         }
 
-        void ReportInvalidMessage(TextMessage message, TextMessageParsingResult parsingResult, DataCollectorId dataCollectorId, bool unknown, CaseReporting caseReporting)
+        void ReportInvalidMessage(
+            TextMessage message,
+            TextMessageParsingResult parsingResult,
+            DataCollectorId dataCollectorId,
+            bool unknown,
```

Comment:
Could maybe named the parameter unknownDataCollector, so many things can be unknown :)

Reply:
Good suggestion 😊